vpr/src/base/vpr_types.h

```diff
@@ -829,36 +829,84 @@ struct t_annealing_sched {
- * place_cost_exp:  Power to which denominator is raised for linear_cong.    *
- * place_chan_width:  The channel width assumed if only one placement is     *
- *                    performed.                                             *
- * pad_loc_type:  Are pins free to move during placement or fixed randomly.  *
- * constraints_file:  File used to lock block locations during placement.    *
- * place_freq:  Should the placement be skipped, done once, or done for each *
- *              channel width in the binary search.                          *
- * recompute_crit_iter: how many temperature stages pass before we recompute *
- *               criticalities based on average point to point delay         *
- * inner_loop_crit_divider: (move_lim/inner_loop_crit_divider) determines how*
- *               many inner_loop iterations pass before a recompute of       *
- *               criticalities is done.                                      *
- * td_place_exp_first: exponent that is used on the timing_driven criticlity *
- *               it is the value that the exponent starts at.                *
- * td_place_exp_last: value that the criticality exponent will be at the end *
- * doPlacement: true if placement is supposed to be done in the CAD flow, false otherwise */
+/******************************************************************
+ * Placer data types
+ *******************************************************************/
+
+/**
+ * @brief Types of placement algorithms used in the placer.
+ *
+ *   @param BOUNDING_BOX_PLACE
+ *              Focuses purely on minimizing the bounding
+ *              box wirelength of the circuit.
+ *   @param CRITICALITY_TIMING_PLACE
+ *              Focuses on minimizing both the wirelength and the
+ *              connection timing costs (criticality * delay).
+ *   @param SLACK_TIMING_PLACE
+ *              Focuses on improving the circuit slack values
+ *              to reduce critical path delay.
+ *
+ * The default is to use CRITICALITY_TIMING_PLACE. BOUNDING_BOX_PLACE
+ * is used when there is no timing information available (wiring only).
+ * SLACK_TIMING_PLACE is mainly feasible during placement quench.
+ */
 enum e_place_algorithm {
     BOUNDING_BOX_PLACE,
-    PATH_TIMING_DRIVEN_PLACE
+    CRITICALITY_TIMING_PLACE,
+    SLACK_TIMING_PLACE
+};
+
+/**
+ * @brief Provides a wrapper around enum e_place_algorithm.
+ *
+ * Supports the method isTimingDriven(), which allows flexible updates
+ * to the placer algorithms if more timing driven placement strategies
+ * are added in tht future. This method is used across various placement
+ * setup files, and it can be useful for major placer routines as well.
+ *
+ * More methods can be added to this class if the placement strategies
+ * will be further divided into more categories the future.
+ *
+ * Also supports assignments and comparisons between t_place_algorithm
+ * and e_place_algorithm so as not to break down previous codes.
+ */
```

Comment:
What is the justification for this over the following?

```
bool is_timing_driven(e_place_algorithm algo) {
  return algo == CRITICALITY_TIMING_PLACE || algo == SLACK_TIMING_PLACE; 
}
```

What methods do you propose should be added to this class? Do they improve the current code? If so, please add them.

Reply:
1. The timing driven check is performed in several places other than ```place.cpp```. So I incorporated this check into ```placer_opts```.
2. There might be future categorizations of placement algorithms/strategies. Currently, there are only two categories (timing and no timing). Having this class will make it easier to describe and implement new categorizations in the future. If you don't think this is necessary, I'll revert the change.